clang/test/Sema/offsetof-ucrt.c

```diff
@@ -0,0 +1,17 @@
+// RUN: %clang_cc1 %s -triple i686-pc-win32 -fsyntax-only -verify -fms-compatibility
+// expected-no-diagnostics
+
+typedef __typeof__(sizeof(0)) size_t;
+
+#if defined _MSC_VER && !defined _CRT_USE_BUILTIN_OFFSETOF
```

Comment:
We should also have a test for `#ifdef _CRT_USE_BUILTIN_OFFSETOF` to make it clear that we're testing that the macro is predefined.